src/blueprint/from_config.rs

```diff
@@ -47,20 +47,71 @@ fn to_directive(const_directive: ConstDirective) -> Valid<Directive> {
 
   Ok(Directive { name: const_directive.name.node.clone().to_string(), arguments, index: 0 })
 }
+
 fn to_schema(config: &Config) -> Valid<SchemaDefinition> {
-  let query = config
-    .graphql
-    .schema
-    .query
-    .as_ref()
-    .validate_some("Query type is not defined".to_string())?;
-
-  Ok(SchemaDefinition {
-    query: query.clone(),
-    mutation: config.graphql.schema.mutation.clone(),
-    directives: vec![to_directive(config.server.to_directive("server".to_string()))?],
-  })
+    let query_type_name = config.graphql.schema.query
+        .as_ref()
+        .validate_some("Query root is missing".to_owned())?;
+
+    // Validate query
+    {
+        let query_type_name = config.graphql.schema.query
+            .as_ref()
+            .validate_some("Query root is missing".to_owned())?;
+
+        let Some(query) = config.find_type(query_type_name) else {
+            return Valid::fail("Query type is not defined".to_owned()).trace(query_type_name);
+        };
+
+        query.fields
+            .iter()
+            .filter(|(_, field)| field.required.is_some())
+            .try_for_each(|(name, field)| {
+                // Validate field has resolver
+                if field.has_resolver() {
+                    Ok(())
+                } else {
+                    Valid::fail("No resolver has been found in the schema".to_owned()).trace(name)
+                }
+            })
+            .trace(query_type_name)?;
+    }
+
+    // Validate mutation
+    {
```

Comment:
use `validate_or` to combine the query and mutation checks together instead of using `?`

Reply:
done